### ISSUE TITLE
[AA4AM] Report on deprecated actions during build pipeline.

### ISSUE THREAD

**Issue by filcole** [microsoft/coe-starter-kit#985]
**Is your feature request related to a problem? Please describe.**
As the connector ecosystem has expanded it is inevitable that some actions within connectors will be deprecated. Some connectors are deprecated totally.  It would be great if a build pipeline would warn (or even fail) when a deprecate action is used in a solution.

**Describe the solution you'd like**
Using the unpacked solution (from source), examine the 
* connection references and flow actions in the power automate json files
* connection reference and powerfx formulas in canvas apps (exploded msapp files)

Once the solution is scanned it could report (or error) on usages of deprecated actions within the solution.

**Describe alternatives you've considered**
This is related to https://github.com/microsoft/coe-starter-kit/issues/981, which aims to aid the maker when maintaining an existing environment.

**Additional context**
I have started work on this and now have:
1. an automatically generated list of deprecated connectors and actions in JSON format, scraped from https://docs.microsoft.com/en-us/connectors/connector-reference/
2. a powershell script that will examine the *.json files in a solution and report on usages of deprecated actions

To complete the proof of concept I plan to expand the powershell script to unpack .msapp files and then search for the connection references and actions.

**Questions**
1. Is powershell the preferred language still, or is node (JS) preferred?

**Comment by filcole** [comment]
@mikefactorial : I have a complete proof of concept PowerShell script written that I'd like to share/commit.  Any thoughts on using PowerShell?

**Comment by mikefactorial** [comment]
Hey @filcole, thanks for the ping. 

I saw the original issue and meant to come back, but lost track. 

We use Powershell pretty extensively still in our pipelines. So, no problem there. 

I'd be happy to take a look at what you have and review it with the team. Would also like to get @devkeydet eyes on it. He's out this week, but back next week. 

NOTE: We are in the process of updating our branching strategy in this repo and in coe-alm-accelerator-templates to remove the vnext branches we've been using so if you create a PR please do so from the main branch until we complete the transition and have the contribution doc updated.

Thanks for the contribution!

**Comment by filcole** [comment]
Hi @mikefactorial and @devkeydet, 

My script is available here: https://github.com/filcole/ConnectorCheck, with an explanatory blog here: https://philcole.org/post/connector-check/.  I'd like to help get this incorporated into AA4AM, perhaps eventually something similar could be included in the OOTB solution checker `pac solution check`.

**Comment by mikefactorial** [comment]
Would like to roll this in with the enhancement here to be able to conditionally call these types of tasks and begin creating a library of conditional extensions that could be opted in to https://github.com/microsoft/coe-starter-kit/issues/1944

**Comment by mikefactorial** [comment]
@filcole better late than never, but we finally got around to reviewing the code in an attempt to include this as a hook in our pipelines. However, due to the reliance on the third-party web api / site we've decided not to add it to our code base. We investigated alternative platform support for querying the information that your code was getting from the website but found nothing comparable that wouldn't take a significant amount of effort to build. Tagging @RajeevPentyala as he did the investigation and can provide more details. Closing for now.
